src/pages/permissions/panels/PermissionSelector.tsx

```diff
@@ -62,39 +71,45 @@ const PermissionSelector: FC<Props> = ({ onAddPermission }) => {
 
   const isLoading = isPermissionsLoading || isMetadataLoading;
 
+  // focus permission selector container on first render so tabbing will consistently navigate to resource type selector first
   useEffect(() => {
-    document.getElementById("resourceType")?.focus();
+    setTimeout(() => {
+      if (permissionSelectorRef.current) {
+        permissionSelectorRef.current.focus();
+        permissionSelectorRef.current.tabIndex = -1;
+      }
+    }, 100);
   }, []);
 
   useEffect(() => {
-    if (!resourceType) {
-      document.getElementById("resourceType")?.focus();
-      return;
-    }
-
-    if (resourceType === "server") {
-      document.getElementById("entitlement")?.focus();
-      return;
+    if (resourceType) {
+      if (resourceType === "server") {
+        entitlementRef.current?.open();
+        return;
+      }
+
+      if (permissions?.length) {
+        resourceRef.current?.open();
+        return;
+      }
+
+      // if there are no permissions, we can't select a resource
+      // keep focus on the resource type selector
+      resourceTypeRef.current?.focus();
     }
-    document.getElementById("resource")?.focus();
   }, [resourceType, permissions]);
 
-  useEffect(() => {
-    if (resource) {
-      document.getElementById("entitlement")?.focus();
-    }
-  }, [resource]);
-
-  useEffect(() => {
-    if (entitlement) {
-      document.getElementById("add-entitlement")?.focus();
-    }
-  }, [entitlement]);
-
   const handleResourceTypeChange = (value: string) => {
     const resourceType = value;
     setEntitlement("");
-    setResourceType(resourceType);
+    setResourceType((prev) => {
+      // in case if the same resource type is selected, open the resource dropdown
+      if (!!resourceType && prev === resourceType && permissions?.length) {
+        resourceRef.current?.open();
+      }
```

Comment:
This shouldn't be a special case. Probably better to remove the useeffect above and always handle the "focus next selector" here.
Because currently there is a special case for server we should go to the next-next selector and we shouldn't have to maintain that special case twice. So generalizing to have one place to handle it directly seems better than the indirect handling with the use effect.

Reply:
The `useEffect` is needed specifically for `resourceType` selection because we need to cater for the case when no permissions (and therefore resources) are available for a resource type. In that case, want to keep the focus on the resource type selector. The permissions for a resource type is fetched from the server when `resourceType` updates so it has to be dealt within a `useEffect`. 

In the case where the `resourceType` selection remains the same, we have to deal with it inside this function since we need to check if the same resource type was selected. Also the `useEffect` above will not trigger.

Reply:
I see so we do need two entry points to cover both cases. Maybe refactor the handler into a function, so we can call it from both entry points.